nodejs/awsx/apigateway/api.ts

```diff
@@ -628,48 +642,80 @@ function addEventHandlerRouteToSwaggerSpec(
+    authorizers: Authorizers[],
+    apiAuthorizers: Record<string, Authorizers>): Record<string, string[]>[] {
 
-    const authNames: string[] = [];
+    const authRecords: Record<string, string[]>[] = [];
     swagger["securityDefinitions"] = swagger["securityDefinitions"] || {};
 
     for (const auth of authorizers) {
         const suffix = Object.keys(swagger["securityDefinitions"]).length;
-        auth.authorizerName = auth.authorizerName || `${swagger.info.title}-authorizer-${suffix}`;
+        const authName = auth.authorizerName || `${swagger.info.title}-authorizer-${suffix}`;
+        auth.authorizerName = authName;
 
         // Check API authorizers - if its a new authorizer add it to the apiAuthorizers
         // if the name already exists, we check that the authorizer references the same authorizer
-        if (!apiAuthorizers[auth.authorizerName]) {
-            apiAuthorizers[auth.authorizerName] = auth;
-        } else if (apiAuthorizers[auth.authorizerName] !== auth) {
-            throw new Error("Two different authorizers using the same name: " + auth.authorizerName);
+        if (!apiAuthorizers[authName]) {
+            apiAuthorizers[authName] = auth;
+        } else if (apiAuthorizers[authName] !== auth) {
+            throw new Error("Two different authorizers using the same name: " + authName);
         }
 
+        let methods: string[] = [];
         // Add security definition if it's a new authorizer
         if (!swagger["securityDefinitions"][auth.authorizerName]) {
-            const securityDef: SecurityDefinition = {
-                type: "apiKey",
-                name: auth.parameterName,
-                in: auth.parameterLocation,
-                "x-amazon-apigateway-authtype": auth.authType,
-                "x-amazon-apigateway-authorizer": getLambdaAuthorizer(auth.authorizerName, auth),
-            };
-            swagger["securityDefinitions"][auth.authorizerName] = securityDef;
+            let securityDef: SecurityDefinition;
+
+            if (lambdaauthorizer.isLambdaAuthorizer(auth)) {
+                securityDef = {
+                    type: "apiKey",
+                    name: auth.parameterName,
+                    in: auth.parameterLocation,
+                    "x-amazon-apigateway-authtype": auth.authType,
+                    "x-amazon-apigateway-authorizer": getLambdaAuthorizer(authName, auth),
+                };
+            } else {
+                securityDef = {
+                    type: "apiKey",
+                    name: auth.parameterName,
+                    in: auth.parameterLocation,
+                    "x-amazon-apigateway-authtype": auth.authType,
+                    "x-amazon-apigateway-authorizer": {
+                        type: auth.authType,
+                        identitySource: lambdaauthorizer.getIdentitySource(auth.identitySource),
+                        providerARNs: getCognitoPoolARNs(auth.providerARNs),
+                    },
+                };
```

Comment:
securityDefs in both branches seem the same (except for the actual authorizer portion).  Consider unifying more of the code and making that the only place where they need to differ.